### PR TITLE
Make current processing rule accessible

### PR DESCRIPTION
**Pull request by xJuvi**
With this PR i want to resolve my bad solution from #598 

I added a new global variable which will contains the current processing rule while processing. So the data (like rule name) are usable in the middleware:

```php
namespace My\Router\Namespace;

use Pecee\Http\Middleware\IMiddleware;
use Pecee\Http\Request;
use Pecee\SimpleRouter\SimpleRouter as Router;

class test implements IMiddleware
{
	public function handle(Request $request) : void
	{
		$router = Router::router();
		$RouteName = $router->getCurrentProcessingRoute()->getName();
	}
}
```

Thanks for merging.

Kind regards

### CODE REVIEW

File: src/Pecee/SimpleRouter/Router.php [hunk 1]
```diff
@@ -371,6 +377,9 @@ public function routeRequest(): ?string
             foreach ($this->processedRoutes as $key => $route) {
 
                 $this->debug('Matching route "%s"', get_class($route));
+                
+                /* Add current processing route to constants */
+                $this->currentProcessingRoute = $route;
```
Comment (DeveloperMarius) [inline src/Pecee/SimpleRouter/Router.php:382]: Why are you initializing `$currentProcessingRoute` as an array, but set and use it like a normal single Route object?
referring to:
 `$router->getCurrentProcessingRoute()->getName();` when return type is an array `getCurrentProcessingRoute(): array`

Comment (xJuvi) [reply src/Pecee/SimpleRouter/Router.php:382]: Oh yeah. It's a bit confused.
I updated my PR. Thanks!

**Comment by Clrkz** [comment]
waiting to merge this good work!

**Comment by xJuvi** [comment]
hi @Clrkz 

i changed the repo to this fork:
https://github.com/DeveloperMarius/simple-php-router/tree/v5-development-php8

There are some fixes for php8 and many other things. The most of open PRs from this project were merged in the forked repo, this PR also.

Kind regards

**Pull request merged**